code/game/objects/items/RPD.dm

```diff
@@ -561,20 +561,27 @@ GLOBAL_LIST_INIT(RPD_recipes, list(
 	. = FALSE
 	switch(p_class) //if we've gotten this var, the target is valid
 		if(PAINT_MODE) //Paint pipes
+			if(!is_paintable)
+				return ..()
 			var/obj/machinery/atmospherics/pipe/P = A
-			playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
-			P.paint(paint_colors[paint_color])
-			user.visible_message("<span class='notice'>[user] paints \the [P] [paint_color].</span>","<span class='notice'>You paint \the [P] [paint_color].</span>")
+			if(P)
```

Comment:
is this check necessary after the `is_paintable` check?

Reply:
Yeah, byond's short-term memory is comparable to schrodinger's cat when it comes to vars like this.

Reply:
actually, better way to do this - remove `is_paintable` and replace that check with its logic. then you can remove this check for P